Libraries/LibWeb/CSS/CSSStyleSheet.h

```diff
@@ -109,11 +111,11 @@ class CSSStyleSheet final : public StyleSheet {
     HashMap<FlyString, GC::Ptr<CSSNamespaceRule>> m_namespace_rules;
     Vector<GC::Ref<CSSImportRule>> m_import_rules;
 
-    GC::Ptr<StyleSheetList> m_style_sheet_list;
     GC::Ptr<CSSRule> m_owner_css_rule;
 
     Optional<URL::URL> m_base_url;
     GC::Ptr<DOM::Document const> m_constructor_document;
+    HashTable<GC::Ptr<DOM::Node>> m_owning_documents_or_shadow_roots;
```

Comment:
This feels very strange. When do we legitimately have multiple owners?

Reply:
Gotcha. Would be good to have a test that covers this as well, then :)

Reply:
FWIW, I believe: http://wpt.live/css/cssom/CSSStyleSheet-constructable.html has a test that adds the same stylesheet to a document and a shadow root, so that test should be improved by this change.

Reply:
sure, I updated the added test to create 2 shadow roots.